packages/entities/src/lib/entity.utils.ts

```diff
@@ -43,3 +46,32 @@ export function findIdsByPredicate<
     predicate(entities[id])
   );
 }
+
+export function findEntityByPredicate<
+  S extends EntitiesState<Ref>,
+  Ref extends EntitiesRef
+>(entities: EntitiesRecord, predicate: ItemPredicate<getEntityType<S, Ref>>) {
+  for (const entity of Object.values(entities)) {
```

Comment:
The `Object.values(entities)` is a redundant operation. Use the same functionality as `findIdsByPredicate`

Reply:
I'm not sure why this is better. Maybe you meant something else.

```ts
export function findEntityByPredicate<S extends EntitiesState<Ref>,
  Ref extends EntitiesRef>(state: S, ref: Ref, predicate: ItemPredicate<getEntityType<S, Ref>>) {
  const { idsKey, entitiesKey } = ref;
  const entities                = state[entitiesKey];

  const id = state[idsKey].find((id: getIdType<S, Ref>) => {
    const entity = entities[id];
    return predicate(entity);
  });
  
  return entities[id] || undefined;
}
```

Reply:
Yes, that's what I meant.

Reply:
Sure. That's more consistent